etl/resources/open_ai_embedding_model_resource.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+
+from dagster import ConfigurableResource
+from langchain.docstore.document import Document
+from langchain.embeddings import CacheBackedEmbeddings
+from langchain.storage import LocalFileStore
+from langchain_community.vectorstores import FAISS
+from langchain_core.embeddings import Embeddings
+from langchain_openai import OpenAIEmbeddings
+
+from etl.models import OpenAiResourceParams, Record
+from etl.models.types import EnrichmentType, RecordType
+
+
+class OpenAiEmbeddingModelResource(ConfigurableResource):  # type: ignore
+    """A ConfigurableResource to create an embedding store with OpenAI's embedding models."""
+
+    openai_resource_params: OpenAiResourceParams
+
+    __output_path = Path(__file__).parent.parent.absolute() / "data" / "output"
+
+    def __create_documents(self, records: tuple[Record, ...]) -> tuple[Document, ...]:
+        """Create and return Documents from Records."""
+
+        if self.openai_resource_params.record_type == RecordType.WIKIPEDIA:
```

Comment:
This code should be in the asset.

Reply:
Yes, I created a new asset that materializes Documents.